internal/manifest/report.go

```diff
@@ -66,17 +72,55 @@ func (r *Report) Add(slice *setup.Slice, fsEntry *fsutil.Entry) error {
 		r.Entries[relPath] = entry
 	} else {
 		r.Entries[relPath] = ReportEntry{
-			Path:   relPath,
-			Mode:   fsEntry.Mode,
-			SHA256: fsEntry.SHA256,
-			Size:   fsEntry.Size,
-			Slices: map[*setup.Slice]bool{slice: true},
-			Link:   fsEntry.Link,
+			Path:       relPath,
+			Mode:       fsEntry.Mode,
+			SHA256:     fsEntry.SHA256,
+			Size:       fsEntry.Size,
+			Slices:     map[*setup.Slice]bool{slice: true},
+			Link:       fsEntry.Link,
+			HardLinkId: hardLinkId,
 		}
 	}
 	return nil
 }
 
+func (r *Report) handleHardlinkReport(fsEntry *fsutil.Entry) (*fsutil.Entry, int) {
```

Comment:
How about this?

```suggestion
func (r *Report) getHardLinkId(fsEntry *fsutil.Entry) (int) {
```

And since `fsEntry` is passed as a pointer, we can update the value from within.